src/encoded/static/components/regulome_search.js

```diff
@@ -759,17 +781,44 @@ class GenomeFacets extends React.Component {
 
     lookupFilterCount(filter, category) {
         const fakeFilter = `${filter}AND${category}`;
+        // for filters that are not selected, we want to display how many new results would be added if that filter were selected
         if (!this.state.selectedFacets.includes(fakeFilter)) {
             // if we add this filter to our selected filters, we need to know how many results there will be
             const fakeFacets = [...this.state.selectedFacets, fakeFilter];
             const fakeFilteredFiles = filterByAllSelectedFilters(this.props.files, fakeFacets);
             // if a filter has already been selected in this category, we do not want to include results for those in our count
-            if (this.state.selectedFacets.filter(d => d.includes(category)).length > 0) {
+            if ((this.state.selectedFacets.filter(d => d.includes(category)).length > 0)) {
+                if (category === 'organ') {
+                    const alreadyMatchingFiles = filterByOneFilter(this.props.filteredFiles, fakeFilter);
+                    return ` (${(fakeFilteredFiles.length - this.props.filteredFiles.length) + alreadyMatchingFiles.length})`;
+                }
                 return ` (${fakeFilteredFiles.length - this.props.filteredFiles.length})`;
             }
             return ` (${fakeFilteredFiles.length})`;
         }
-        return '';
+        // for filters that are already selected, we want to display how many results that are displayed match this filter
+        // we have an extra space here so that selected terms do not match the string pattern that other terms do, so that if a selected term has a (0), it will not get filtered out as a zero, it will stay in the top category with non-zero matching terms
+        const matchingFiles = filterByOneFilter(this.props.filteredFiles, fakeFilter);
+        return `  (${matchingFiles.length})`;
+    }
+
+    isntZero(arr, category) {
+        return arr.filter(el => this.lookupFilterCount(el, category) !== ' (0)');
+    }
+
+    isZero(arr, category) {
+        return arr.filter(el => this.lookupFilterCount(el, category) === ' (0)');
+    }
+
+    placeZerosAtEnd(arr, facet) {
+        let arrBegin = [...arr];
+        arrBegin = this.isntZero(arrBegin, facet).sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()));
```

Comment:
It looks to me like you could pass `arr` to `this.isntZero` instead of making a copy in `arrBegin` first because `this.isntZero` makes a filtered copy of its first parameter anyway, and doesn’t mutate anything. Same thing for `arrEnd` below.